lib/node_modules/@stdlib/math/base/special/bessely1/test/test.js

```diff
@@ -0,0 +1,268 @@
+	var x;
+	var y;
+	var i;
+
+	expected = smaller.expected;
+	x = smaller.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 4.5 * EPS * abs( expected[i] );
+			t.equal( delta <= tol, true, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. Δ: '+delta+'. Tolerance: '+tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the Bessel function of the second kind of one order (Y_1) (tiny positive values)', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+
+	expected = tinyPositive.expected;
+	x = tinyPositive.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 8.0 * EPS * abs( expected[i] );
+			t.equal( delta <= tol, true, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. Δ: '+delta+'. Tolerance: '+tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'The Bessel function of the second kind of one order (Y_1) (subnormal values)', function test( t ) {
+	var x;
+	var y;
+	var i;
+
+	x = subnormal.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		t.equal( y, NINF, 'returns -Infinity');
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the Bessel function of the second kind of one order (Y_1) (huge positive values)', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+
+	expected = hugePositive.expected;
+	x = hugePositive.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 4500.0 * EPS * abs( expected[i] );
+			t.equal( delta <= tol, true, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. Δ: '+delta+'. Tolerance: '+tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the Bessel function of the second kind of order one (Y_1) for positive x over a wide range of magnitudes', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+
+	expected = positiveGamut.expected;
+	x = positiveGamut.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 16.0 * EPS * abs( expected[i] );
+			t.equal( delta <= tol, true, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. Δ: '+delta+'. tol: '+tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function returns `NaN` for negative numbers', function test( t ) {
+	var v;
+	var x;
+	var i;
+
+	for ( i = 0; i < 1000; i++ ) {
+		x = -randu() * 100.0;
```

Comment:
`randu()` can return `0`. Subtract `float64-eps` to ensure never `0`.

Reply:
Nice catch!